posts/schrodingers-pineapple-upside-down-cake-front-end-development-for-back-end-engineers.md

```diff
@@ -0,0 +1,367 @@
+---
+authors:
+- Christina Annas
+tags:
+-back-end
+-career-development
+-front-end
+-javascript
+date: 2022-02-28T22:00:00.000Z
+title: "Schrödinger's Pineapple Upside Down Cake: Front-End Development for Back-End Engineers"
+---
+
+Hey, good to see you again! Let's start by reviewing where we left off. We have a webpage that displays weather information for each of the Ippon offices in the US. However, although our page is created in a much more dynamic way than how it started out, we are still displaying out-of-date information. 
+
+```
+todo: add in from-last-time screenshot here.
+```
+
+It's the moment we've all been waiting for -- let's use current data! If you're coming to front-end development from a back-end perspective, like I did, you may already know that we'll be fetching the current weather data by making an HTTP request to an API endpoint. As a reminder, there is an accompanying [GitHub repository](https://github.com/christinaannas/weather-at-the-office) to follow along with this series. 
+
+## Our API
+
+We'll be getting our data from a API at `http://api.weatherapi.com`, which requires an free API key. Instructions for getting your own API key are in the README in the GitHub repo we've been using. However, so that you don't have to request one in order to follow along, we'll use a straightforward little Node.js server as a bit of middleware between our front end and that API. Our server looks for a `WEATHER_API_KEY` environment variable and will use it to call the API if present, but otherwise will pull data from some cached responses for each of our four cities. It also tries the cached data in case the API call fails. I've copied a few details in the listing below, but left some bits out so we can focus on a couple of aspects at play.
```

Comment:
a free API key

Reply:
This sounds a little clunky, but I don't have a rec to fix

> However, so that you don't have to request one in order to follow along

Reply:
Maybe something like:

>  However, we'll use a straightforward little Node.js server as a bit of middleware between our front end and the API to get our key.

Reply:
Yeah let me rework that. Probably need to split out a sentence or reorder some stuff... Maybe something like 

> We'll be getting our data from an API at `http://api.weatherapi.com`, which requires a free API key. We'll use a straightforward little Node.js server as a bit of middleware between our front end and that API. Our server looks for a `WEATHER_API_KEY` environment variable and will use it to call the API if present. Don't worry, though -- you can still follow along if you don't want to request an API key. Our server accounts for this by pulling data from some cached responses for each of our four cities if API key is not present or if the API call fails. I've copied a few details in the listing below, but left some bits out so we can focus on a couple of aspects at play. 

How does that read for you?